snowflake-usage-monitor-cf.yaml

```diff
@@ -144,56 +144,81 @@ Resources:
+              else:
+                  print("Teams message posted successfully")
 
           def handler(event, context):
-            result = []
-            status_code = 200
-            try:
-                event_body = json.loads(event["body"])
-                rows = event_body["data"]
-
-                for row in rows:
-                    database, forecast, change = row[1:]
-                    result.append(f"*{database}* - Forecast: {forecast} ({change.strip()})")
-                
-                secret = get_secret("slackurl", "us-east-1")
-                
-                slack_url = json.loads(secret)["slack_url"]
-                send_slack_message(slack_url, '\n'.join(result))
-                
-                teams_url = json.loads(secret)["teams_url"]
-                send_teams_message(teams_url, '\n'.join(result))
-
-            except Exception as err:
-                status_code = 400
-
-            return {"statusCode": status_code}
+              result = []
+              status_code = 200
+              try:
+                  event_body = json.loads(event["body"])
+                  rows = event_body["data"]
+                  
+                  account_width = 19
+                  forecast_width = 12
+                  
+                  for idx, row in enumerate(rows):
+                      if idx == 0:
+                          database, forecast, change = row[1:]
+                          header = f"{'Account'.ljust(account_width)} | {'Forecast'.ljust(forecast_width)} | {'Change'.strip()}"
+                          result.append(header)
+                          divider = "-" * len(header)
+                          result.append(divider)
+
+                      database, forecast, change = row[1:]
+                      if not database.endswith("(GB)"):
+                          forecast = f"${forecast}"
+                      else:
+                          forecast = str(forecast)
+                      str_forecast = str(forecast).ljust(forecast_width)
+                      result.append(f"{database.ljust(account_width)} | {str_forecast} | {change.strip()}")
+                  
+                  secret = get_secret(SECRET_MANAGER, REGION)
+                  
+                  slack_url = json.loads(secret)[SLACK_SECRET_KEY_NAME]
+                  send_slack_message(slack_url, '\n'.join(result))
+                  
+                  teams_url = json.loads(secret)[TEAMS_SECRET_KEY_NAME]
+                  send_teams_message(teams_url, '\n'.join(result))
+
+              except Exception as err:
+                  status_code = 400
+
+              return {"statusCode": status_code}
       Description: "Echo Lambda created from CloudFormation template"
       FunctionName: !Ref "lambdaName"
       Handler: "index.handler"
       Role: !GetAtt "lambdaExecutionIAMRole.Arn" 
-      Runtime: "python3.9"
+      Runtime: "python3.7"
```

Comment:
is this python version correct?

Reply:
yes. requests library is deprecated in 3.9, and so i downgraded to 3.7